File: Behavioral/Strategy/CSharp/CreditCardPayment.cs
// CreditCardPayment.cs

using System;

public class CreditCardPayment : IPaymentStrategy {
  private string _cardNumber;

  public CreditCardPayment(string cardNumber) {
    _cardNumber = cardNumber;
  }

  public void Pay(double amount) {
    Console.WriteLine($"Paid ${amount} using Credit Card (Card Number: {_cardNumber})");
  }
}

File: Behavioral/Strategy/Cpp/CreditCardPayment.cpp
// CreditCardPayment.cpp

#include "CreditCardPayment.h"

CreditCardPayment::CreditCardPayment(const std::string& cardNumber) : cardNumber(cardNumber) {}

void CreditCardPayment::pay(double amount) {
  std::cout << "Paid $" << amount << " using Credit Card (Card Number: " << cardNumber << ")" << std::endl;
}
